libs/datasets/statistical_areas.py

```diff
@@ -103,4 +105,47 @@ def from_local_data() -> "CountyToHSAAggregator":
         hsa_raw_map = dict(hsa_df.values)
         return CountyToHSAAggregator(county_map=hsa_raw_map)
 
-    # TODO(sean): Add an aggregate() function here when aggregating hospital data to HSAs
+    def aggregate(
+        self,
+        dataset_in: MultiRegionDataset,
+        fields_to_aggregate: List[CommonFields] = [
+            CommonFields.STAFFED_BEDS,
+        ],  # add all the hospital fields here
+    ) -> MultiRegionDataset:
+
+        # Only aggregate county-level data
+        counties = dataset_in.get_subset(aggregation_level=AggregationLevel.COUNTY)
+        fields_to_keep = [
+            col for col in counties.timeseries.columns if col not in fields_to_aggregate
+        ]
+        ds_in_ts: pd.DataFrame = counties.timeseries.loc[:, fields_to_aggregate]
+        ds_ts_other: pd.DataFrame = counties.timeseries.loc[:, fields_to_keep]
+
+        ds_in_ts[CommonFields.HSA] = ds_in_ts.index.get_level_values("location_id").map(
+            self.county_to_hsa_region_map
+        )
+
+        # Calculate HSA-level data only for selected fields.
+        hsa_ts: pd.DataFrame = (
+            ds_in_ts.groupby([CommonFields.HSA, CommonFields.DATE])
+            .sum()
+            .loc[:, fields_to_aggregate]
+            .reset_index()
+        )
+
+        # Map HSA data to each county
+        ds_in_ts = ds_in_ts.loc[:, [CommonFields.HSA]].reset_index()
+        ds_in_ts = ds_in_ts.merge(hsa_ts, on=[CommonFields.DATE, CommonFields.HSA])
+
+        # Set and sort the index to group locations to make index monotonic increasing.
+        # Index locations became unordered after unsetting and resetting the index.
+        ds_in_ts = ds_in_ts.set_index([CommonFields.LOCATION_ID, CommonFields.DATE]).sort_index()
+
+        # Combine aggregated fields with untouched fields.
+        out_ts = ds_in_ts.join(ds_ts_other)
+
+        # Drop all the columns used in the calculations and combine with non-county data.
+        out_ts = out_ts.loc[:, counties.timeseries.columns]
+        out_ts = out_ts.combine_first(dataset_in.timeseries)
+
+        return dataclasses.replace(counties, timeseries=out_ts, timeseries_bucketed=None)
```

Comment:
We might want to update the tags here to clarify that this data has been calculated/aggregated